To: vim_dev@googlegroups.com
Subject: Patch 7.3.272
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.272
Problem:    ":put =list" does not add an empty line for a trailing empty
            item.
Solution:   Add a trailing NL when turning a list into a string.
Files:      src/eval.c
    

*** ../vim-7.3.271/src/eval.c	2011-07-15 21:24:06.000000000 +0200
--- src/eval.c	2011-08-10 12:31:51.000000000 +0200
***************
*** 1357,1363 ****
--- 1357,1367 ----
  	{
  	    ga_init2(&ga, (int)sizeof(char), 80);
  	    if (tv.vval.v_list != NULL)
+ 	    {
  		list_join(&ga, tv.vval.v_list, (char_u *)"\n", TRUE, 0);
+ 		if (tv.vval.v_list->lv_len > 0)
+ 		    ga_append(&ga, NL);
+ 	    }
  	    ga_append(&ga, NUL);
  	    retval = (char_u *)ga.ga_data;
  	}
*** ../vim-7.3.271/src/version.c	2011-08-10 12:19:00.000000000 +0200
--- src/version.c	2011-08-10 12:36:41.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     272,
  /**/

-- 
Lose weight, NEVER Diet again with
                  The "Invisible Weight Loss Patch"
						(spam e-mail)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
